tests/ip/test_ip_packet.py

```diff
@@ -0,0 +1,379 @@
+import logging
+import re
+
+import pytest
+import ipaddress
+import time
+import random
+import ptf.testutils as testutils
+from ptf import mask, packet
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.portstat_utilities import parse_portstat
+
+
+class TestIPPacket(object):
+    PKT_NUM = 1000
+    PKT_NUM_MIN = PKT_NUM * 0.9
+    PKT_NUM_MAX = PKT_NUM * 1.3
+
+    def skip(self, mg_facts):
+        if len(mg_facts["minigraph_interfaces"]) < 2 and len(mg_facts["minigraph_portchannels"]) < 2:
+            pytest.skip("Skipping ip packet test since can't provide enough interfaces")
+
+    @staticmethod
+    def sum_portstat_ifaces_counts(portstat_out, ifaces, column):
+        if len(ifaces) == 0:
+            return 0
+        if len(ifaces) == 1:
+            return int(portstat_out[ifaces[0]][column].replace(",", ""))
+        return sum(map(lambda iface: int(portstat_out[iface][column].replace(",", "")), ifaces))
+
+    @staticmethod
+    def parse_interfaces(output_lines, pc_ports_map):
+        """
+        Parse the inerfaces from 'show ip route' into an array
+        """
+        route_targets = []
+        ifaces = []
+        output_lines = output_lines[3:]
+
+        for item in output_lines:
+            match = re.search("(Ethernet\d+|PortChannel\d+)", item)
+            if match:
+                route_targets.append(match.group(0))
+
+        for route_target in route_targets:
+            if route_target.startswith("Ethernet"):
+                ifaces.append(route_target)
+            elif route_target.startswith("PortChannel") and route_target in pc_ports_map:
+                ifaces.extend(pc_ports_map[route_target])
+
+        return ifaces
+
+    @pytest.fixture(scope="class")
+    def test_param(self, duthost, tbinfo):
```

Comment:
Please check https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices/sonic.py#L1256. stats parsing should be available in that file too.